virl2_client/virl2_client.py

```diff
@@ -805,14 +813,31 @@ def join_existing_lab(self, lab_id: str, sync_lab: bool = True) -> Lab:
         self._labs[lab_id] = lab
         return lab
 
-    def get_diagnostics(self) -> dict:
+    def get_diagnostics(
+        self, categories: list[DiagnosticCategory] | None = None
+    ) -> dict:
         """
-        Return the controller diagnostic data as a JSON object.
+        Return selected diagnostic data as a JSON object.
 
+        :param categories: List of diagnostic categories to fetch. If None, fetch all.
         :returns: The diagnostic data.
         """
-        url = self._url_for("diagnostics")
-        return self._session.get(url).json()
+        if categories is None:
+            categories = list(DiagnosticCategory)
```

Comment:
perhaps issue a warning that one should select categories. and the categories can be a varargs parameter.